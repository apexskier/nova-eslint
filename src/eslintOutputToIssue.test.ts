import type { Linter } from "eslint";
import { eslintOutputToIssue } from "./eslintOutputToIssue";

(global as any).IssueSeverity = {
  Info: "info",
  Warn: "warn",
  Error: "error",
};

(global as any).Issue = class MockIssue {
  readonly args: ReadonlyArray<any>;
  constructor(...args: Array<any>) {
    this.args = args;
  }
};

describe("eslintOutputToIssue", () => {
  test.each([0, 1, 2])("severity %d", (severity) => {
    expect(
      eslintOutputToIssue({
        column: 1,
        line: 2,
        endColumn: 3,
        endLine: 4,
        ruleId: "rule_id",
        message: "message",
        messageId: "message_id",
        nodeType: "node_type",
        severity: severity as Linter.Severity,
        fix: { text: "fix_text", range: [5, 6] },
        suggestions: [
          {
            desc: "suggestion description",
            fix: { text: "suggestion_fix_text", range: [7, 8] },
          },
        ],
      })
    ).toMatchSnapshot();
  });
});
