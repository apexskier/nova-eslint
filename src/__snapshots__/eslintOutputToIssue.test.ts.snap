// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslintOutputToIssue severity 0 1`] = `
MockIssue {
  "args": Array [],
  "code": "rule_id",
  "column": 1,
  "endColumn": 3,
  "endLine": 4,
  "line": 2,
  "message": "message",
  "severity": "info",
  "source": "ESLint",
}
`;

exports[`eslintOutputToIssue severity 1 1`] = `
MockIssue {
  "args": Array [],
  "code": "rule_id",
  "column": 1,
  "endColumn": 3,
  "endLine": 4,
  "line": 2,
  "message": "message",
  "severity": undefined,
  "source": "ESLint",
}
`;

exports[`eslintOutputToIssue severity 2 1`] = `
MockIssue {
  "args": Array [],
  "code": "rule_id",
  "column": 1,
  "endColumn": 3,
  "endLine": 4,
  "line": 2,
  "message": "message",
  "severity": "error",
  "source": "ESLint",
}
`;
